src/components/Comments.jsx

```diff
@@ -1,17 +1,46 @@
 import { useEffect, useState } from "react";
-import { getComments } from "../../api";
+import { getComments, postComment } from "../../api";
 
 export const Comments = ({article_id}) => {
     const [comments, setComments] = useState([])
+    const [commentInput, setCommentInput] = useState("")
     
     useEffect(() => {
+        // if(newComment != ""){
+        //     postComment(article_id, newComment).then((data) => setComments(data, ...comments))
+        //     setNewComment("");        
+        // }
         getComments(article_id).then((data) => {
             setComments(data)
         })
-    },[article_id])
+    },[article_id, comments])
+
+    const handleChange = (e) => {
+        setCommentInput(e.target.value);
+      };
+    
+    const handleSubmit = (e) => {
+        e.preventDefault();
+        const newComment = commentInput;
+        setComments([newComment, ...comments]) 
+        setCommentInput("");
+        postComment(article_id, newComment).then((data) => setComments(data, ...comments))
+        console.log(comments)
```

Comment:
Once you are happy things are working, logs need to be remove

Reply:
I'd recommend a refactor of how you're setting comments after a post. the setComments should either be receiving an array or a function that creates an array. 

These comments aren't being optimistically rendered as you have added comments to the dependancy array the whole page reloads when posting a comment. The comment is then taken from the top of the list and added to the bottom (this will no doubt be the default order in your BE project)